app/restore_to_file.go

```diff
@@ -59,6 +64,78 @@ func RestoreToFileCmd() cli.Command {
 	}
 }
 
+type Restore struct {
+	sync.RWMutex
+	SnapshotName    string
+	RestoreProgress int
+	RestoreError    error
+	LastUpdatedAt   time.Time
+}
+
+func (r *Restore) UpdateRestoreStatus(snapshot string, rp int, re error) {
+	r.Lock()
+	defer r.Unlock()
+
+	r.SnapshotName = snapshot
+	r.RestoreProgress = rp
+	r.RestoreError = re
+	r.LastUpdatedAt = time.Now()
+}
+
+func restore(url string) error {
+	restoreObj := &Restore{}
+
+	backupURL := util.UnescapeURL(url)
+	config := &backupstore.DeltaRestoreConfig{
+		BackupURL: backupURL,
+		DeltaOps:  restoreObj,
+		Filename:  BackupFilePath,
+	}
+
+	err := backupstore.RestoreDeltaBlockBackup(config)
+	if err != nil {
+		return err
+	}
+
+	bar := pb.StartNew(100)
+	periodicChecker := time.NewTicker(Retry * time.Second)
+	progress := 0
+
+	for _ = range periodicChecker.C {
+		restoreObj.RLock()
```

Comment:
Get the values from restoreObj then release the lock. It's easier this way since you're only reading from the object, not writing into it.